public/javascripts/averages.js

```diff
@@ -1,69 +0,0 @@
-// Establish empty object with the keys that we're going to use to calculate averages.
-
-// Set the key array length to 0, so we can increment it later
-
-// Create a function that will calculate the averages of a set of reviews we give it. Within this function we will loop through the keys (overallScore, features etc) of the object we established above.
-
-// We loop through each review using the first key (overallScore), looking for the value. We use filter and map to create an array of those values. Eg. Overall Scores = [3,4,3]. If for some reason, there are keys that have not been scored yet, we will skip to the next key from our averages object.
-
-// Then, we take the array we created in the last step and run a reduce on it to get the sum, and then the average. We then insert the average of that key (overallScore) into the averages object we set at the start of the function.
-
-// Once we have looped through all of the keys in averages, we will have an object that contains their average score for each of the keys present.
-
-// The function ultimately returns the average of all metrics, across all the reviews left for a company, so will be our headline number.
-
-const internalCalcs = (reviews) => {
```

Comment:
Generally, this is far too complicated and difficult to read. Always, always remember you're writing code for other developers to read. For me, looking at this, this is definitely not code I want to read. 

I'd add line breaks, split this into several parts, abstract some logic out to other utility functions, add comments, use more descriptive variable names (`keyArrayL` is not a good name, very hard to immediately discover what it's for, I'm not sure why we're returning `average` and `averages` at the end)

Overall, I'd take another go at this. Before you do, have a think about exactly what you want to do. Make a list, write some pseudo-code, before diving into it

Reply:
Also `internalCalcs` gives me no indication of what this function does. If you compare to English, functions are verbs, they're _doing_ things. They should be named as verbs (`fetchData`, `renderString`, etc). They should also be descriptive of what the function does. If you find it hard to name it, there's a good chance your function is doing too much and should be abstracted out

Reply:
### totally, I knew this one was the worst bit and held together with duct tape! I'll have another crack at it.